File: programozas/vending_machine.c
#include <stdio.h>

typedef struct {
    char name[20];
    int price;
} Drink;

Drink drinks[] = {
    {"Kávé", 250},
    {"Tea", 200},
    {"Forró csoki", 300},
    {"Víz", 150}
};

void print_drinks() {
    printf("Válassz egy italt:\n");
    for (int i = 0; i < 4; i++) {
        printf("%d: %s - %d Ft\n", i + 1, drinks[i].name, drinks[i].price);
    }
}   

int choose_drink() {
    int choice;
    printf("Add meg a választott ital számát: ");
    scanf("%d", &choice);

    while (choice < 1 || choice > 4) {
        printf("Érvénytelen választás.\n");
        printf("Add meg a választott ital számát: ");
        scanf("%d", &choice);
    }
    return choice;           
}

void pay_drink(int choice) {
    int price = drinks[choice - 1].price;
    char *drink_name = drinks[choice - 1].name;
    int inserted_money;

    printf("%s választva. Ár: %d Ft\n", drink_name, price);
    printf("Dobd be a pénzt: ");
    scanf("%d", &inserted_money);

    if (inserted_money < price) {
        printf("Nincs elég pénzed az italhoz.\n");
    } else if (inserted_money > price) {
        printf("Itt a %s, és a visszajáró: %d Ft.\n", drink_name, inserted_money - price);
    } else {
        printf("Itt a %s. Köszönjük a vásárlást!\n", drink_name);
    }	
}
    
void vending_machine() {
    while (1) {
        printf("Üdvözöl az italautomata!\n");
        print_drinks();
        int choice;
        choice = choose_drink();
        pay_drink(choice);
    }
}

int main() {
    vending_machine();
    return 0;
}  
                        

File: programozas/vending_machine.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const drinks = {
    1: { name: "Kávé", price: 250 },
    2: { name: "Tea", price: 200 },
    3: { name: "Forró csoki", price: 300 },
    4: { name: "Víz", price: 150 }
};

function printDrinks() {
    console.log("Válassz egy italt:");
    for (const [key, drink] of Object.entries(drinks)) {
        console.log(`${key}: ${drink.name} - ${drink.price} Ft`);
    }
}

function askQuestion(question) {
    return new Promise((resolve) => {
        rl.question(question, (answer) => resolve(answer));
    });
}

async function chooseDrink() {
    while (true) {
        const choice = await askQuestion("Add meg a választott ital számát: ");
        if (drinks[choice]) {
            return choice;
        } else {
            console.log("Érvénytelen választás.");
        }
    }
}

async function payDrink(choice) {
    const { name, price } = drinks[choice];
    console.log(`${name} választva. Ár: ${price} Ft`);
    
    const insertedMoney = parseInt(await askQuestion("Dobd be a pénzt: "), 10);
    
    if (insertedMoney < price) {
        console.log("Nincs elég pénzed az italhoz.");
    } else if (insertedMoney > price) {
        console.log(`Itt a ${name}, és a visszajáró: ${insertedMoney - price} Ft.`);
    } else {
        console.log(`Itt a ${name}. Köszönjük a vásárlást!`);
    }
}

async function vendingMachine() {
    while (true) {
        console.log("Üdvözöl az italautomata!");
        printDrinks();
        const choice = await chooseDrink();
        await payDrink(choice);
    }
}

vendingMachine();
